pkg/controller/bookkeepercluster/bookie.go

```diff
@@ -209,15 +211,15 @@ func MakeBookieConfigMap(pravegaCluster *v1alpha1.PravegaCluster) *corev1.Config
 		"-XX:HeapDumpPath=" + heapDumpDir,
 	}
 
-	if match, _ := util.CompareVersions(pravegaCluster.Spec.Version, "0.4.0", ">="); match {
-		// Pravega < 0.4 uses a Java version that does not support the options below
+	if match, _ := util.CompareVersions(bookkeeperCluster.Spec.Version, "0.4.0", ">="); match {
```

Comment:
this logic doesn't apply here. 0.4 is for Pravega version. Perhaps Bookeeper doesn't need this logic at all if they are always using newer version of Java.

Reply:
Good catch. Will remove this check.